src/libs/actions/Session/index.js

```diff
@@ -137,45 +102,42 @@ function resendValidationLink(login = credentials.login) {
  *
  * @param {String} login
  */
-function fetchAccountDetails(login) {
-    Onyx.merge(ONYXKEYS.ACCOUNT, {...CONST.DEFAULT_ACCOUNT_DATA, loading: true});
+function beginSignIn(login) {
+    const optimisticData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: ONYXKEYS.ACCOUNT,
+            value: {
+                ...CONST.DEFAULT_ACCOUNT_DATA,
+                isLoading: true,
+            },
+        },
+    ];
 
-    DeprecatedAPI.GetAccountStatus({email: login, forceNetworkRequest: true})
-        .then((response) => {
-            if (response.jsonCode === 200) {
-                Onyx.merge(ONYXKEYS.CREDENTIALS, {
-                    login: response.normalizedLogin,
-                });
-                Onyx.merge(ONYXKEYS.ACCOUNT, {
-                    accountExists: response.accountExists,
-                    validated: response.validated,
-                    closed: response.isClosed,
-                    forgotPassword: false,
-                    validateCodeExpired: false,
-                });
-
-                if (!response.accountExists) {
-                    createAccount(login);
-                } else if (response.isClosed) {
-                    reopenAccount(login);
-                } else if (!response.validated) {
-                    resendValidationLink(login);
-                }
-            } else if (response.jsonCode === 402) {
-                Onyx.merge(ONYXKEYS.ACCOUNT, {
-                    error: ValidationUtils.isNumericWithSpecialChars(login)
-                        ? Localize.translateLocal('common.error.phoneNumber')
-                        : Localize.translateLocal('loginForm.error.invalidFormatEmailLogin'),
-                });
-            } else if (response.jsonCode === CONST.JSON_CODE.UNABLE_TO_RETRY) {
-                Onyx.merge(ONYXKEYS.ACCOUNT, {error: Localize.translateLocal('session.offlineMessageRetry')});
-            } else {
-                Onyx.merge(ONYXKEYS.ACCOUNT, {error: response.message});
-            }
-        })
-        .finally(() => {
-            Onyx.merge(ONYXKEYS.ACCOUNT, {loading: false});
-        });
+    const successData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: ONYXKEYS.ACCOUNT,
+            value: {
+                isLoading: false,
+            },
+        },
+    ];
+
+    const failureData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: ONYXKEYS.ACCOUNT,
+            value: {
+                isLoading: false,
+                errors: {
+                    [DateUtils.getMicroseconds()]: 'Cannot get account details, please try again',
```

Comment:
Just saw this randomly. This is wrong, this is a string that should be localized. Can you send a PR to fix that please?

Reply:
Do you have a link for the context for this? These updates to `errors` have been breakneck - I want to make sure I have everything straight.

Reply:
Oh, wait I see it. Yeah, my bad, updating!